core/src/main/java/tech/tablesaw/io/xlsx/XlsxReader.java

```diff
@@ -0,0 +1,313 @@
+                        altColumn.append((int) num);
+                        return altColumn;
+                    } else if ((long) num == num) {
+                        Column<Long> altColumn = LongColumn.create(column.name(), column.size());
+                        altColumn = shortColumn.mapInto(s -> (long) s, altColumn);
+                        altColumn.append((long) num);
+                        return altColumn;
+                    } else {
+                        Column<Double> altColumn = DoubleColumn.create(column.name(), column.size());
+                        altColumn = shortColumn.mapInto(s -> (double) s, altColumn);
+                        altColumn.append(num);
+                        return altColumn;
+                    }
+                } else if (column.type() == ColumnType.INTEGER) {
+                    Column<Integer> intColumn = (Column<Integer>) column;
+                    if ((int) num == num) {
+                        intColumn.append((int) num);
+                        return null;
+                    } else if ((long) num == num) {
+                        Column<Long> altColumn = LongColumn.create(column.name(), column.size());
+                        altColumn = intColumn.mapInto(s -> (long) s, altColumn);
+                        altColumn.append((long) num);
+                        return altColumn;
+                    } else {
+                        Column<Double> altColumn = DoubleColumn.create(column.name(), column.size());
+                        altColumn = intColumn.mapInto(s -> (double) s, altColumn);
+                        altColumn.append(num);
+                        return altColumn;
+                    }
+                } else if (column.type() == ColumnType.LONG) {
+                    Column<Long> longColumn = (Column<Long>) column;
+                    if ((long) num == num) {
+                        longColumn.append((long) num);
+                        return null;
+                    } else {
+                        Column<Double> altColumn = DoubleColumn.create(column.name(), column.size());
+                        altColumn = longColumn.mapInto(s -> (double) s, altColumn);
+                        altColumn.append(num);
+                        return altColumn;
+                    }
+                } else if (column.type() == ColumnType.DOUBLE) {
+                    Column<Double> doubleColumn = (Column<Double>) column;
+                    doubleColumn.append(num);
+                    return null;
+                }
+            }
+            break;
+        case BOOLEAN:
+            if (column.type() == ColumnType.BOOLEAN) {
+                Column<Boolean> booleanColumn = (Column<Boolean>) column;
+                booleanColumn.append(cell.getBooleanCellValue());
+                return null;
+            }
+        default:
+            break;
+        }
+        return null;
+    }
+
+    public Column<?> createColumn(String name, Cell cell) {
```

Comment:
Can we move this method to the test since it looks like it's only used there? Or else make it `protected` instead of `public`